src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1292,4 +1299,1344 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                companionCoilFieldTag = "companion_cooling_heat_pump_name";
+                refCapFieldTag = "nominal_heating_capacity";
+            }
+
+            // A1-A3
+            thisPLHP.name = UtilityRoutines::MakeUPPERCase(thisObjectName);
+            std::string loadSideInletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("water_inlet_node_name").get<std::string>());
+            std::string loadSideOutletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("water_outlet_node_name").get<std::string>());
+            // Implicit
+            // std::string condenserType = "AIRSOURCE"; // UtilityRoutines::MakeUPPERCase(fields.at("condenser_type").get<std::string>());
+            thisPLHP.waterSource = false;
+            thisPLHP.airSource = false;
+            // A4
+            std::string sourceSideInletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("air_source_node_name").get<std::string>());
+            std::string sourceSideOutletNodeName =
+                "DUMMY_CONDENSER"; // UtilityRoutines::MakeUPPERCase(fields.at("source_side_outlet_node_name").get<std::string>());
+            srand(time(NULL));
+            for (int i = 1; i <= 3; ++i) {
+                sourceSideOutletNodeName += ("_" + std::to_string(rand()));
+            }
+
+            // A5
+            if (fields.find(companionCoilFieldTag) != fields.end()) { // optional field
+                thisPLHP.companionCoilName = UtilityRoutines::MakeUPPERCase(fields.at(companionCoilFieldTag).get<std::string>());
+            }
+
+            // A6 Fuel Type
+            std::string tempRsrStr = UtilityRoutines::MakeUPPERCase(fields.at("fuel_type").get<std::string>());
+            thisPLHP.fuelType = DataGlobalConstants::AssignResourceTypeNum(tempRsrStr);
+            // Validate fuel type input
+            static constexpr std::string_view RoutineName("processInputForEIRPLHP: ");
+            bool FuelTypeError = false;
+            UtilityRoutines::ValidateFuelTypeWithAssignResourceTypeNum(tempRsrStr, tempRsrStr, thisPLHP.fuelType, FuelTypeError);
+            if (FuelTypeError) {
+                ShowSevereError(state, format("{}{}=\"{}\",", RoutineName, cCurrentModuleObject, thisPLHP.name));
+                ShowContinueError(state, format("Invalid Fuel Type = {}", tempRsrStr));
+                ShowContinueError(state, "Reset the Fuel Type to \"NaturalGas\".");
+                thisPLHP.fuelType = DataGlobalConstants::ResourceType::Natural_Gas;
+                errorsFound = true;
+                FuelTypeError = false;
+            }
+
+            // A7 End use category
+            thisPLHP.endUseSubcat = UtilityRoutines::MakeUPPERCase(fields.at("end_use_subcategory").get<std::string>());
+            if (thisPLHP.endUseSubcat == "") {
+                thisPLHP.endUseSubcat = "Heat Pump Fuel Fired"; // or "General"?
+            }
+
+            // N1 Nominal heating capacity
+            auto tmpRefCapacity = fields.at(refCapFieldTag);
```

Comment:
So, whereas `find` returns an iterator, `at` returns an object reference so need to use `auto &` here.

Reply:
Addressed in earlier changes.